sstables/mx/reader.cc

```diff
@@ -132,25 +132,12 @@ class data_consume_rows_context_m : public data_consumer::continuous_data_consum
     enum class state {
         PARTITION_START,
         DELETION_TIME,
-        DELETION_TIME_3,
         FLAGS,
         FLAGS_2,
-        EXTENDED_FLAGS,
-        CK_BLOCK_HEADER,
-        CK_BLOCK_END,
-        ROW_BODY_TIMESTAMP_DELTIME,
-        ROW_BODY_DELETION_3,
-        ROW_BODY_MISSING_COLUMNS_2,
-        ROW_BODY_MISSING_COLUMNS_READ_COLUMNS_2,
-        COLUMN,
-        NEXT_COLUMN,
-        COLUMN_TIMESTAMP,
-        COLUMN_DELETION_TIME_2,
-        COLUMN_TTL_2,
-        COLUMN_END,
         OTHER,
     } _state = state::PARTITION_START;
 
+    bool _consuming = true;
```

Comment:
In a previous lifetime, I was the one who introduced this "non_consuming" thing, but I don't remember why - so it would be nice to have a comment explanaing this variable. I have a vague recollection that in some case we had states that would still need to keep changing without consuming any new data. But does your code also have this? Can't your code just continue to modify the state without yielding anything, and only yield if it needs to read more data?

Reply:
To be honest, I also weren't sure about the exact meaning of it, so I only assured that after the changes it's "syntactically" consistent with the former version.
However, after simplifying it, it turns out we return true for ```non_consuming()``` when we needed to return from the coroutine (for examble because the buffer is full) although there is **some** point in code that we can reach without reading new data, where we prepare another mutation_fragment (using some ```consume*``` method from ```_consumer```).

It was harder to get the meaning before the patch, because we were also checking some but not all states where we got ```read_status::waiting``` preceding the states where we use ```consume*```, and it was still working because if we have ```read_status::waiting``` we don't even check ```non_consuming()```

> Can't your code just continue to modify the state without yielding anything, and only yield if it needs to read more data?

As mentioned above, we need to return from the coroutine even if we read enough to keep working, so we most likely still need to have a variable for non_consuming